### DISCUSSION TITLE
소스 코드 타입에 따른 실행 컨텍스트 차이를 설명해주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#70] [실행컨텍스트]
- 소스 코드 타입에 따른 실행 컨텍스트 차이를 설명해주세요.

**Comment by publdaze** [comment]
> - 소스 코드 타입에 따른 실행 컨텍스트 차이를 설명해주세요.
- 전역 코드 - 전역 코드 평가 시 전역 변수 관리를 위한 전역 스코프가 생성되고, var로 선언된 변수 또는 함수선언문은 전역 객체의 프로퍼티와 메서드가 되어 렉시컬 환경을 이루고 전역 실행컨텍스트에 바인딩 됩니다.
- 함수 코드 - 함수 코드 평가 시 매개변수, 지역 변수, arguments 객체 관리를 위한 지역 스코프가 생성되고, 상위 렉시컬 환경에 대한 참조가 바인딩 되어 렉시컬 환경을 이루고 함수 실행 컨텍스트에 바인딩 됩니다.
- eval 코드 - eval 코드 평가 시 strict mode에서 독자적인 스코프가 생성되어 렉시컬 환경을 이루고 eval 실행 컨텍스트에 바인딩 됩니다.
- 모듈 코드 - 모듈별로 독립적인 모듈 스코프가 생성되어 렉시컬 환경을 이루고 모듈 실행 컨텍스트에 바인딩 됩니다.

**Comment by pipisebastian** [comment]
- 4가지로 나뉘며, 타입에 따라 실행컨텍스트 관리 내용이 차이가 있습니다.

- `전역 코드` (전역 실행컨텍스트)
    - 최상위 스코프, 전역 스코프 생성
    - var 선언 전역변수, 함수선언문으로 정의한 전역 함수 관리 → 전역객체와 바인딩
- `함수 코드` (지역 실행컨텍스트)
    - 지역 스코프 생성
    - 매개변수, 지역변수, arguments 객체 관리
- `eval 코드` (eval 실행컨텍스트)
    - strict mode에서 독자적인 스코프 생성.
- `모듈 코드` (모듈 실행컨텍스트)
    - 모듈 스코프 생성.
